src/box/lua/console.lua

```diff
@@ -481,20 +497,28 @@ local text_connection_mt = {
                 -- Make sure it is exactly "\set output" command.
                 if operators[items[1]] == set_param and
                     param_handlers[items[2]] == set_output then
-                    local err, fmt = parse_output(items[3])
+                    local err, fmt, _, eos = parse_output(items[3])
                     if not err then
                         self.fmt = fmt
                         self.eos = output_eos[fmt]
+                        if eos ~= nil then
+                            self.local_lua_eos = eos
+                            -- Crop string, so 'eos=.' is not sent to server.
+                            return text:sub(1, text:find(',eos=') - 1)
```

Comment:
It seems, you forgot to change the option name here and no test case fails. We definitely lack of a proper test case here.

BTW, I think that attempt to cut an option from a string is fragile anyway. I would rebuild the command based on parsed command:

```lua
@@ -494,15 +492,26 @@ local text_connection_mt = {
                 -- Make sure it is exactly "\set output" command.
                 if operators[items[1]] == set_param and
                     param_handlers[items[2]] == set_output then
-                    local err, fmt, _, eos = parse_output(items[3])
+                    local err, fmt, opts, local_eos = parse_output(items[3])
                     if not err then
+                        -- Track server-side output format and EOS
+                        -- on the client. It is important to know,
+                        -- where to stop read a response.
                         self.fmt = fmt
                         self.eos = output_eos[fmt]
-                        if eos ~= nil then
-                            self.local_lua_eos = eos
-                            -- Crop string, so 'eos=.' is not sent to server.
-                            return text:sub(1, text:find(',eos=') - 1)
+
+                        -- Set client-side EOS, which is used to
+                        -- generate output for a user on the
+                        -- client (to replace server-side EOF).
+                        if local_eos ~= nil then
+                            self.local_lua_eos = local_eos
                         end
+
+                        -- Rebuild the command without
+                        -- local_eos=<...>, because it is a client
+                        -- side option.
+                        local server_cmd_opts = table.concat({fmt, opts}, ',')
+                        return ('\\set output %s'):format(server_cmd_opts)
                     end
                 end
             end
```

Reply:
After all, I think I agree that reconstruction is a better way to do it. Done.